azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/functions/CleanupResources.java

```diff
@@ -58,36 +69,67 @@ public CleanupResources(AzureComputeApi azureComputeApi,
+            boolean deploymentDeleteStatus = false;
+
+            if (jobDone) {
+               StorageService ss = api.getStorageAccountApi(group).get(storageAccountName);
+               if (ss != null) {
+                  storageAcctDeleteStatus = api.getStorageAccountApi(group).delete(storageAccountName);
+               } else {
+                  storageAcctDeleteStatus = true;
+               }
+               Deployment deployment = api.getDeploymentApi(group).get(id);
+               if (deployment != null) {
                   uri = api.getDeploymentApi(group).delete(id);
                   jobDone = resourceDeleted.apply(uri);
                   if (jobDone) {
-                     // Delete public ip
-                     boolean ipDeleteStatus = api.getPublicIPAddressApi(group).delete(id + "publicip");
-
-                     // Delete Virtual network
-                     boolean vnetDeleteStatus = api.getVirtualNetworkApi(group).delete(group + "virtualnetwork");
-                     return ipDeleteStatus && vnetDeleteStatus;
-
+                     deploymentDeleteStatus = true;
+                  }
+               } else {
+                  deploymentDeleteStatus = true;
+               }
+               NetworkInterfaceCard nic = api.getNetworkInterfaceCardApi(group).get(id + "nic");
+               if (nic != null) {
+                  uri = api.getNetworkInterfaceCardApi(group).delete(id + "nic");
+                  if (uri != null) {
+                     jobDone = resourceDeleted.apply(uri);
+                     if (jobDone) {
+                        boolean ipDeleteStatus = false;
+                        PublicIPAddress ip = api.getPublicIPAddressApi(group).get(id + "publicip");
+                        if (ip != null) {
+                           ipDeleteStatus = api.getPublicIPAddressApi(group).delete(id + "publicip");
+                        } else {
+                           ipDeleteStatus = true;
+                        }
+
+                        // Get NSG
+                        boolean nsgDeleteStatus = false;
+                        NetworkSecurityGroup nsg = api.getNetworkSecurityGroupApi(group).get(id + "nsg");
+                        if (nsg != null) {
+                           uri = api.getNetworkSecurityGroupApi(group).delete(id + "nsg");
+                           jobDone = resourceDeleted.apply(uri);
+                           if (jobDone) {
+                              nsgDeleteStatus = true;
+
+                           }
+                        }
+                        else {
+                           nsgDeleteStatus = true;
+                        }
+
+                        return deploymentDeleteStatus && storageAcctDeleteStatus && ipDeleteStatus && nsgDeleteStatus;
+                     } else {
+                        return false;
+                     }
                   } else {
                      return false;
```

Comment:
Hmmm can we have some way of determining if we can remove it? For example, when the last VM is being removed, etc? Otherwise jclouds will start leaking resources users will potentially be unaware of.